### DISCUSSION TITLE
Support for eCash (XEC)

### DISCUSSION THREAD

**Discussion by 3point14159** [horizontalsystems/unstoppable-wallet-android#5368] [Ideas]
Devland: https://www.bitcoinabc.org/CONTRIBUTING.html

Abouts:

https://www.bitcoinabc.org/ecash/

https://coinmarketcap.com/currencies/ecash/

Do you know of any other Bitcoin forks that are integrating hybrid [PoW](https://e.cash/blog/xec-ecash-mining)/[PoS](https://www.bitcoinabc.org/2022-09-07-avalanche-staking-tutorial/) by using [in-house Avalanche subnets](https://avalanche.cash/)? eCash is led by the [lead developer](https://www.youtube.com/watch?v=c6nVO6rI4D4) from Bitcoin Cash (BCH), this latest iteration (eCash/BitcoinABC) is meant to sustainably fund a CBDC-competitive infrastructure that retains user autonomy.

Latest development snapshot:

https://twitter.com/eCashOfficial/status/1565011254779019264

**Comment by 3point14159** [comment]
Is there a better place to post discussion topics where they are actually discussed? Seems like no one actually reads anything posted here.

**Reply by esengulov** [reply]
Thanks, sorry for delayed reply. We are yet to get usded to track discussions as they were activated just recently. 

As for ecash, we haven't considered yet but based on what you have mentioned adding them should ve failry easy. To be honest we are not reakky familiar with that specific project and not clear on whether there is a substantial demand for it just yet.

**Reply by 3point14159** [reply]
Well, the project has much more market volume/velocity than most of the coins you already support, so that should show you whatever "substantial demand" you aren't seeing. If not, just pretend it is an NFT, if market volume doesn't do it for you, to get around to adding it to the kit, but please fix your non-functional wallet first. People still cannot use the wallet with entering their complex BIP39 passwords over a month after reporting the bug. The vision of this wallet sounds promising, but you guys really need to attract some talent to the dev team; the update process here is terrible.

**Reply by esengulov** [reply]
> Well, the project has much more market volume/velocity than most of the coins you already support, so that should show you whatever "substantial demand" you aren't seeing. If not, just pretend it is an NFT, if market volume doesn't do it for you, to get around to adding it to the kit, but please fix your non-functional wallet first. People still cannot use the wallet with entering their complex BIP39 passwords over a month after reporting the bug. The vision of this wallet sounds promising, but you guys really need to attract some talent to the dev team; the update process here is terrible.

@Dianaaiym could you add eCash integration research to 0.28. @esen could you look into it

**Reply by 3point14159** [reply]
Thank you for taking a look into it. Should be pretty straightforward. If there are any issues, the devs are pretty responsive.

**Comment by 3point14159** [comment]
Can we add this to the ToDo list yet? It would be great to be one of the few wallets to support this coin in the market at the moment. The coin has been operational since 2020, and it headed by the former lead developer of Bitcoin Cash. Should be very simple to integrate.

**Comment by 3point14159** [comment]
I see you guys are actively researching Tron:

https://github.com/horizontalsystems/unstoppable-wallet-android/issues/3618

Any chance we can get eCash on the To Do list? It should plug-in pretty easy. Let's be a leader in the market with this blockchain.

**Reply by esengulov** [reply]
Yeah, TRON was heavily requested among support emails. We spent some time researching it but dropped it due to a number of customization on its side. The development costs and added complexities to the UI would far outweigh its benefits.

As for eCash we plan on researching it in 0.31 and if it's compatible with current architecture then it should be easy to add it.

**Reply by esengulov** [reply]
https://github.com/horizontalsystems/unstoppable-wallet-android/issues/5720

**Reply by 3point14159** [reply]
@esengulov Makes sense, gotta throw a cost-benefit analysis at everything, else you'd be chasing dragons instead of adding functionality.

eCash should be as easy as adding BCH, you may even be able to add it on a .30 release. Happy coding! Oh, when is HS/UW going to implement the PRO version or a donations page (or in-app)?

**Comment by serrq** [comment]
XEC seems to be a great project. I hope dev team bring eCash blockchain in UW family.
I would like have it too.

**Reply by 3point14159** [reply]
@serrq eCash is looking to really be pushing the envelope on the performance of blockchains in a unique way utilizing Avalanche protocol (not their blockchain) for <5 second 1st-block confirmations. There are also much security and privacy benefits to appreciate. Of all the BTC forks, I think these guys are really going to provide the best market solution as use-cases and adoption ramp-up past the all-time highs.

Considering how easy it is to integrate, it will give the users of UW ground-floor access to a high performance blockchain that has a lot of gas in the tank. Cheers to the HS coders here and eCash team for pioneering new territory.

**Reply by serrq** [reply]
What is average transaction finality speed?

**Reply by 3point14159** [reply]
@serrq Perceptively instant. Wanna test if/when it is enabled in UW?

**Reply by 3point14159** [reply]
Here is their landing page for displaying the metrics of their blockchain:

https://avalanche.cash/

Looks like it is right on the horizon for their roadmap.

From their website:

Instant Transaction Finality
Allowing all businesses (including exchanges) to reduce their deposit confirmation requirement to 0
Post-consensus finalizes transactions after a single block is found, due to reorg being prevented by the Avalanche consensus. Pre-consensus validates individual transactions before they are included in a block, reducing confirmation time for a transaction from 10 minutes to a few seconds - that’s more than 100 times faster than Proof of Work-only cryptos like Bitcoin.

Future - feature is coming as part of a later milestone

**Reply by serrq** [reply]
So it has Bitcoin strong security and Avalanche near istant transaction finality and consensus mechanism.

**Reply by 3point14159** [reply]
@serrq That's the best way to sum it up in one sentence. Genius is simplicity.

**Reply by serrq** [reply]
> @serrq Perceptively instant. Wanna test if/when it is enabled in UW?

Of course! 

Do you have a Matrix address to stay in contact?

**Reply by 3point14159** [reply]
I don't Matrix. Do you Session?

**Reply by serrq** [reply]
Just removed, sorry. :(

**Comment by 3point14159** [comment]
An update for anyone reading these threads: eCash has been integrated into Unstoppable Wallet!

https://github.com/horizontalsystems/unstoppable-wallet-android/issues/5962#event-8923134086

https://github.com/horizontalsystems/unstoppable-wallet-android/issues/5363#issuecomment-1501923538

https://github.com/horizontalsystems/unstoppable-wallet-android/issues/5431#event-8921848257

A big thanks to the team for all the efforts! May more coins join HS-UW in the near future.
